internal/scanner/scanner.go

```diff
@@ -36,71 +42,69 @@ type ResourcesFetcher interface {
 type Scanner struct {
 	policiesFetcher  PoliciesFetcher
 	resourcesFetcher ResourcesFetcher
+	reportStore      report.PolicyReportStore
 	// http client used to make requests against the Policy Server
 	httpClient http.Client
 }
 
 // NewScanner creates a new scanner with the PoliciesFetcher provided
-func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher) *Scanner {
-	return &Scanner{policiesFetcher, resourcesFetcher, http.Client{}}
+func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher) (*Scanner, error) {
+	report, err := report.NewPolicyReportStore()
+	if err != nil {
+		return nil, err
+	}
+	return &Scanner{policiesFetcher, resourcesFetcher, *report, http.Client{}}, nil
 }
 
 // ScanNamespace scans resources for a given namespace
-func (s *Scanner) ScanNamespace(namespace string) error {
-	log.Info().Str("namespace", namespace).Msg("scan started")
+func (s *Scanner) ScanNamespace(nsName string) error {
+	log.Info().Str("namespace", nsName).Msg("scan started")
 
-	policies, err := s.policiesFetcher.GetPoliciesForANamespace(namespace)
+	policies, skippedNum, err := s.policiesFetcher.GetPoliciesForANamespace(nsName)
 	if err != nil {
 		return err
 	}
-	log.Debug().Str("namespace", namespace).Int("count", len(policies)).Msg("number of policies to evaluate")
-
-	// TODO continue with the scanning and remove this code
-	log.Debug().Str("namespace", namespace).Msg("The following policies were found for the namespace " + namespace)
-	for _, policy := range policies {
-		log.Debug().Str("policy name", policy.GetName()).Msg("Policy retrieved")
+	namespace, err := s.policiesFetcher.GetNamespace(nsName)
+	if err != nil {
+		return err
```

Comment:
nitpicking, maybe we should look for the `Namespace` first, then look for the other resources defined inside of it

Reply:
True, done in https://github.com/kubewarden/audit-scanner/pull/40/commits/5fc30be0cad27129eda6b3c4a38e10a336d5cc22.